PGM/src/main/java/tc/oc/pgm/shop/currency/MaterialCurrency.java

```diff
@@ -0,0 +1,83 @@
+package tc.oc.pgm.shop.currency;
+
+import com.google.common.util.concurrent.AtomicDouble;
+import net.md_5.bungee.api.chat.BaseComponent;
+import org.bukkit.Material;
+import tc.oc.pgm.features.FeatureDefinition;
+import tc.oc.pgm.match.MatchPlayer;
+import tc.oc.pgm.shop.purchasable.Purchasable;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A currency based on a {@link Material}.
+ */
+public class MaterialCurrency extends FeatureDefinition.Impl implements Currency {
+
+    private final Material material;
+    private final BaseComponent nameSingle;
+    private final BaseComponent namePlural;
+    private final double value;
+
+    public MaterialCurrency(Material material, BaseComponent nameSingle, BaseComponent namePlural, double value) {
+        this.material = material;
+        this.nameSingle = nameSingle;
+        this.namePlural = namePlural;
+        this.value = value;
+    }
+
+    @Override
+    public BaseComponent getSingularName() {
```

Comment:
If this has access to `getValue`, why not have one `getName` method and have it figure out which key to use?